src/model/page-config-resolver.js

```diff
@@ -292,6 +335,120 @@ class MicrodataParser {
     this.doc_ = doc;
   }
 
+  /**
+   * Returns true if access is restricted, otherwise false
+   * @param {Node} root A node that is an item of type 'NewsArticle'
+   * @return {?boolean} locked access
+   * @private
+   */
+  discoverLocked_(root) {
+    const ALREADY_SEEN = 'alreadySeenForAccessInfo';
+    const nodeList = root
+        .querySelectorAll("[itemprop='isAccessibleForFree']");
+    for (let i = 0; nodeList[i]; i++) {
+      const element = nodeList[i];
+      const content = element.content || element.textContent;
+      if (!content) {
+        continue;
+      }
+      if (content.toLowerCase() == 'true') {
+        value = true;
+      } else if (content.toLowerCase() == 'false') {
+        value = false;
+      }
+      if (this.isValidElement_(element, root, ALREADY_SEEN)) {
+        return !value;
```

Comment:
This is kind of hard to read: if element is valid, reverse the value? It'd probably be better to just call it `locked` and calculate it accordingly.

Reply:
addressed.